### PR TITLE
[FX-5690] Migrate PageSidebar to TailwindCSS

### PR DESCRIPTION
**Pull request by sashuk**
[FX-5690]

### Description

This pull request migrates `PageSidebar` to TailwindCSS.

### How to test

- [Temploy](https://picasso.toptal.net/fx-5690-pagesidebar/?path=/story/components-pagesidebar--pagesidebar)
- Check `PageSidebar` in different screens and browsers

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [N/A] Double check if `picasso-tailwind-merge` requires major update (check its `README.md`)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [N/A] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [N/A] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5690]: https://toptal-core.atlassian.net/browse/FX-5690?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/base/Page/src/PageSidebar/PageSidebar.tsx [hunk 1]
```diff
@@ -101,10 +104,33 @@ export const PageSidebar = forwardRef<HTMLDivElement, Props>(function Sidebar(
...
       style={style}
-      className={cx(classes.root, className, classes[variant], classes[size], {
-        [classes.rootCollapsed]: collapsible && isCollapsed,
-        [classes.hamburgerNotAvailable]: !hasTopBar,
-      })}
+      className={twMerge(
+        'h-full',
+        'shadow-[inset_-1px_0_0_0]',
+        'shadow-graphite-800',
+        'text-lg/[inherit]',
```
Comment (sashuk) [inline packages/base/Page/src/PageSidebar/PageSidebar.tsx:111]: `[inherit]` is needed not to interfere line height that affects menu spacing and visuals

File: packages/base/Page/src/PageSidebar/PageSidebar.tsx [hunk 2]
```diff
@@ -101,10 +104,33 @@ export const PageSidebar = forwardRef<HTMLDivElement, Props>(function Sidebar(
...
+        className,
+        classesByVariant[variant],
+        classesBySize[size],
+        'xs:max-md:overflow-y-scroll',
+        collapsible &&
+          isCollapsed && [
+            'w-[5rem]',
+            '[&]:before:w-[5.75rem]',
+            '[&_.scrollable-content]:[-ms-overflow-style:none]',
+            '[&_.scrollable-content]:[scrollbar-width:none]',
```
Comment (sashuk) [inline packages/base/Page/src/PageSidebar/PageSidebar.tsx:130]: The `&::-webkit-scrollbar` is missing (from [here](https://github.com/toptal/picasso/blob/bdcc9b0005b912a50bec7f9c26539c61e15a8e27/packages/base/Page/src/PageSidebar/styles.ts#L82)), but it does not influence the browser and is not standard https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar

File: packages/base/Page/src/PageSidebar/PageSidebar.tsx [hunk 3]
```diff
@@ -101,10 +104,33 @@ export const PageSidebar = forwardRef<HTMLDivElement, Props>(function Sidebar(
...
+        '[&]:before:absolute',
+        '[&]:before:content-[""]',
+        '[&]:before:left-0',
+        '[&]:before:top-0',
+        '[&]:before:w-[15.5rem]',
+        '[&]:before:h-full',
+        className,
+        classesByVariant[variant],
+        classesBySize[size],
+        'xs:max-md:overflow-y-scroll',
```
Comment (sashuk) [inline packages/base/Page/src/PageSidebar/PageSidebar.tsx:124]: The `xs:max-md:w-[100vw]` is missing as it does not affect anything, confirming it in https://toptal-core.slack.com/archives/CCC3GP6CC/p1728473840157329

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]

File: packages/base/Page/src/PageSidebar/PageSidebar.tsx [hunk 4]
```diff
@@ -101,10 +104,33 @@ export const PageSidebar = forwardRef<HTMLDivElement, Props>(function Sidebar(
...
       ref={ref}
       style={style}
-      className={cx(classes.root, className, classes[variant], classes[size], {
-        [classes.rootCollapsed]: collapsible && isCollapsed,
-        [classes.hamburgerNotAvailable]: !hasTopBar,
-      })}
+      className={twMerge(
+        'h-full',
+        'shadow-[inset_-1px_0_0_0]',
+        'shadow-graphite-800',
```
[the remark this answers is not in the document]
Comment (sashuk) [reply packages/base/Page/src/PageSidebar/PageSidebar.tsx:110]: @TomasSlama Deleted, thank you!

**Review by OleksandrNechai — APPROVED** [review]
Looks good 👍

**Pull request merged**